javascript/javascript1/week4/homework/script.js

```diff
@@ -0,0 +1,80 @@
+let myName = "";
+const todo = [];
+
+function getReply(command) {
+
+    if (command.startsWith("Hello my name is") && myName === "") {
+        const myArr = command.split(" ");
+        myName = myArr[myArr.length - 1];
+        return `Nice to meet you ${myName}`;
+    } else if (command.includes("Hello my name is") && myName !== "") {
+        return `${myName} Hi again`;
+    } else if (command === "What is my name?") {
+        if (typeof myName === "string" && myName !== "") {
+            return `Your name is ${myName}`;
+        } else {
+            return `you haven\'t introduce yourself`;
+        }
+    } else if (command.startsWith("Add") && command.endsWith("to my todo")) {
+        let itemToAdd = command.replace("Add", "").replace("to my todo", "");
+        todo.push(itemToAdd);
+        return `${itemToAdd} added to your todo!`;
+    } else if (command.startsWith("Remove") && command.endsWith("from my todo")) {
+        let itemToRemove = command.replace("Remove", "").replace("from my todo", "");
+        for (let i = 0; i < todo.length; i++) {
```

Comment:
It is a good solution, but I recommend that use higher order functions next time.

Reply:
Thanks for your recommendation Vahab. sure I'll try to use from next time.